mlflow/cli.py

```diff
@@ -128,7 +129,12 @@ def ui(file_store, host, port):
     The UI will be visible at http://localhost:5000 by default.
     """
     # TODO: We eventually want to disable the write path in this version of the server.
-    mlflow.server._run_server(file_store, file_store, host, port, 1)
+    try:
+        mlflow.server._run_server(file_store, file_store, host, port, 1)
+    except ShellCommandException:
+        print("There was an issue running the mlflow server. Please see the logs above for details.",
```

Comment:
I also considered making the exception message more specific than `"Non-zero exitcode: 1` and printing the exception message instead, but I felt what I came up with was still not as user-friendly as the text here.